api/src/main/java/org/openmrs/api/context/UserContext.java

```diff
@@ -79,28 +79,31 @@
 	 */
 	private Integer locationId;
 	
+	/**
+	 * The authentication scheme for this user
+	 */
+	private AuthenticationScheme authenticationScheme;
+	
 	/**
 	 * Default public constructor
 	 */
-	public UserContext() {
```

Comment:
Do we still have a default public constructor?

Reply:
@dkayiwa the only public constructor that remains takes an auth. scheme as argument. This has been around for a few commits yet. `UserContext` is always instantiated via `new` (it's never Spring-wired). All places where this is done should provide the auth. scheme from `Context`:
```java
new UserContext(Context.getAuthenticationScheme())
```